sbin/jtreg_prep.sh

```diff
@@ -0,0 +1,129 @@
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Purpose: This script was designed to do any+all setup required by the jtreg.sh script in order to run it.
+# Tasks: Retrieve Java, unpack it if need-be, and store it locally in a specific location. If the location is blank, we put it in 
+
+REPOSITORY=AdoptOpenJDK/openjdk-jdk8u
+OPENJDK_REPO_NAME=openjdk
+
+while [[ $# -gt 0 ]] && [[ ."$1" = .-* ]] ; do
+  opt="$1";
+  shift;
+  case "$opt" in
+    "--" ) break 2;;
+
+    "--java_source" )
+    JAVA_SOURCE="$1"; shift;;
+
+    "--java_destination" )
+    JAVA_DESTINATION="$1"; shift;;
+
+    "--ssh" | "-S" )
+    USE_SSH=true; shift;;
+
+    "--repository" | "-r" )
+    REPOSITORY="$1"; shift;;
+
+    "--branch" | "-b" )
+    BRANCH="$1"; shift;;
+
+    "--working_dir" )
+    WORKING_DIR="$1"; shift;;
+
+    *) echo >&2 "Invalid option: ${opt}"; echo "This option was unrecognised. See the script jtreg_prep.sh for a full list."; exit 1;;
+   esac
+done
+
+if [ -z "${JAVA_SOURCE}" ]; then
+  echo >&2 "jtreg_prep.sh failed: --java_source must be specified"; exit 1
+fi
+
+if [ -z "${WORKING_DIR}" ] ; then
+  echo "WORKING_DIR is undefined so setting to $PWD"
+  WORKING_DIR=$PWD
+else
+  echo "Working dir is $WORKING_DIR"
+fi
+
+if [ -z "${JAVA_DESTINATION}" ]; then
+  JAVA_DESTINATION="$WORKING_DIR/$OPENJDK_REPO_NAME/build/java_home/images"
+fi
+
+if [ -z "${BRANCH}" ] ; then
+  echo "BRANCH is undefined so checking out dev"
+  BRANCH="dev"
+fi
+
+# Step 1: Fetch OpenJDK, as that's where the tests live.
+
+if [ -d "$WORKING_DIR"/$OPENJDK_REPO_NAME/.git ] && [ $REPOSITORY == "AdoptOpenJDK/openjdk-jdk8u" ] ; then
+  # It does exist and it's a repo other than the AdoptOpenJDK one
+  cd $WORKING_DIR/$OPENJDK_REPO_NAME
+  echo "Will reset the repository at $PWD in 10 seconds..."
+  sleep 10
+  echo "Pulling latest changes from git repo"
+  git fetch --all
+  git reset --hard origin/$BRANCH
+  cd $WORKING_DIR
+elif [ ! -d "${WORKING_DIR}"/$OPENJDK_REPO_NAME/.git ] ; then
+  # If it doesn't exixt, clone it
+  echo "Didn't find any existing openjdk repository at WORKING_DIR (set to ${WORKING_DIR}) so cloning the source to openjdk"
+  if [[ "${USE_SSH}" == true ]] ; then
+    echo "git clone -b ${BRANCH} git@github.com:${REPOSITORY}.git $WORKING_DIR/$OPENJDK_REPO_NAME"
+    git clone -b ${BRANCH} git@github.com:${REPOSITORY}.git $WORKING_DIR/$OPENJDK_REPO_NAME
+  else
+    echo "git clone -b ${BRANCH} https://github.com/${REPOSITORY}.git $WORKING_DIR/$OPENJDK_REPO_NAME"
+    git clone -b ${BRANCH} https://github.com/${REPOSITORY}.git $WORKING_DIR/$OPENJDK_REPO_NAME
```

Comment:
General nit, there doesn't seem to be a reason to use `${}` in some places and just `$` in others in this script. Maybe just drop the `{}`.